### DISCUSSION TITLE
Subfilter with Exclude Item (negate sign)

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6469] [Q&A]
* open 
* create a new tiddler with below wikitext
```
Example i

><$vars xx="-[[HelloThere]] [[GettingStarted]] [[Community]] ">

<$list filter="[[HelloThere]] [subfilter<xx>]">

</$list>
</$vars>


Example ii

><$list filter="[[HelloThere]] -[[HelloThere]] [[GettingStarted]] [[Community]]">

</$list>
```

You get

Example i

> [HelloThere](https://tiddlywiki.com/#HelloThere)
> [GettingStarted](https://tiddlywiki.com/#GettingStarted)
> [Community](https://tiddlywiki.com/#Community)

> Example ii
> 
> [GettingStarted](https://tiddlywiki.com/#GettingStarted)
> [Community](https://tiddlywiki.com/#Community)

Why negate sign does not work in subfilter?

**Comment by pmario** [comment]
The first example are 2 filter runs. Try this `<$list filter="[[HelloThere]subfilter<xx>]"/>`

~~In the first example, the subfilter internally is called like this `[[HelloThere]] [all[tiddlers]subfilter<xx>]` ... That's probably an example, where our convenience functions "trick" us.~~

Edit: having a look at the source -- there is no `all[tiddlers]`

**Comment by kookma** [comment]
Thank you @pmario, so in

> `<$list filter="[[HelloThere]] [subfilter<xx>]">`

the subfilter is first calculated and the result is used as a filter with `[[HelloThere]]`.

**Reply by pmario** [reply]
Filters are evaluated from left -> right. ... I think, there is a problem with the syntax: `[[` and `]]` are **not** like a link in wikitext. 

`[[HelloThere]] [subfilter<xx>]` is the complete [filter expression](https://tiddlywiki.com/#Filter%20Expression) with [2 filter runs](https://tiddlywiki.com/#Filter%20Run)

`[[HelloThere]]` ... Is the "short form" of `[title[HelloThere]]`  ...  it's the first run
`[subfilter<xx>]` ... Is the second run

The 2 runs are combined as described in [filter expression](https://tiddlywiki.com/#Filter%20Expression) with `run run` which is short form of `[run] :or[run]` ... which is "de-duplicated **union** of sets"

IMO the important thing in your OP first example `<$list filter="[[HelloThere]] [subfilter<xx>]">` is that the 2 runs need to be seen separated. 

`[[HelloThere]]` produces `HelloThere`
`[subfilter<xx>]` produces `GettingStarted Community`

and then the 2 results are "united"

Edit: This post was heavily edited. .. But I think it is valid now

**Comment by pmario** [comment]
Your first example does work as expected if you use: `<$list filter="[[HelloThere]] +[subfilter<xx>]">` which is `[run] +[run]` ... **accumulation of filter steps** ... as described in the docs. 

~~So I'm not sure about `[all[tiddlers]subfilter<xx>]` anymore. I'll have to have a look at the source code.~~

**Reply by pmario** [reply]
I think I got it right now. ... hmmm ... It is complex ;)

**Comment by kookma** [comment]
Thank you Mario!
